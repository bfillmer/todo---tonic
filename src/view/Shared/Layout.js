
import styled from 'styled-components'

export const Section = styled.section`
  width: 33rem;
  margin: 0 auto;
`
