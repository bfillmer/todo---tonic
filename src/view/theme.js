
const richBlack = `rgba(4, 15, 15, 1)`
const platinum = `rgba(226, 239, 222, 1)`
const greenMunsell = `rgba(13, 171, 118, 1)`
const internationalOrange = `rgba(193, 41, 46, 1)`

export const theme = {
  colors: {
    primary: platinum,
    secondary: greenMunsell,
    black: richBlack,
    warning: internationalOrange
  },
  bottomMargin: '1.5rem'
}

// Simple helper function, takes in any number of props mapping to properties within the theme
// object and returns the value.
export const getTheme = (...props) => ({theme}) => props.reduce((t, p) => t[p], theme)
