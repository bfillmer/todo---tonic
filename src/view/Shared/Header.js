
import React from 'react'
import styled from 'styled-components'

import {H1} from 'view/Shared/Typography'

const Wrapper = styled.header`
  text-align: center;
  padding-top: 2rem;
  padding-bottom: 2rem;
`

// HEADER COMPOSITION
export const Header = ({name}) => (
  <Wrapper>
    <H1>T&T</H1>
  </Wrapper>
)
