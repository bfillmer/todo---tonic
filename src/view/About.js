
import React from 'react'
import Link from 'redux-first-router-link'

import {routeHome} from 'actions'

import {Section} from 'view/Shared/Section'
import {H1, P} from 'view/Shared/Typography'

export const About = () => (
  <Section>
    <H1>About</H1>
    <P>Head to the <Link to={routeHome()}>Home Page</Link>.</P>
  </Section>
)
