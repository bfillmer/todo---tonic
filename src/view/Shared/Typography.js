
import styled from 'styled-components'

import {getTheme} from 'view/theme'

export const H1 = styled.h1`
  margin-bottom: ${getTheme('bottomMargin')};
`

export const P = styled.p`
  margin-bottom: ${getTheme('bottomMargin')};
`
