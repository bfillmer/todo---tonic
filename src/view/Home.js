
import React from 'react'
import Link from 'redux-first-router-link'

import {routeAbout} from 'actions'

import {Section} from 'view/Shared/Section'
import {H1, P} from 'view/Shared/Typography'

export const Home = () => (
  <Section>
    <H1>Home</H1>
    <P>Head to the <Link to={routeAbout()}>About Page</Link>.</P>
  </Section>
)
