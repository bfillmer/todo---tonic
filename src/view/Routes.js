
import React from 'react'
import {connect} from 'react-redux'
import {NOT_FOUND} from 'redux-first-router'

import {routeType} from 'selectors'
import {ROUTE_HOME, ROUTE_ABOUT} from 'types'
import {About} from 'view/About'
import {Home} from 'view/Home'

const routesMap = {
  [ROUTE_HOME]: Home,
  [ROUTE_ABOUT]: About,
  [NOT_FOUND]: Home
}

const mapStateToProps = state => ({
  route: routeType(state)
})

const Container = ({route}) => {
  const Route = routesMap[route] ? routesMap[route] : routesMap[NOT_FOUND]
  return (
    <Route />
  )
}

export const Routes = connect(mapStateToProps)(Container)
