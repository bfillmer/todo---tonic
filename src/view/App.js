
import React from 'react'
import styled, {ThemeProvider} from 'styled-components'

import {Header} from 'view/Shared/Header'
import {Routes} from 'view/Routes'

import {theme} from 'view/theme'

// GLOBAL WRAPPER
const Wrapper = styled.div`
  height: 100vh;
`

export const App = () => (
  <ThemeProvider theme={theme}>
    <Wrapper>
      <Header />
      <Routes />
    </Wrapper>
  </ThemeProvider>
)
