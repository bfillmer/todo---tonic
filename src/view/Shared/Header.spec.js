/* global expect, test */

import React from 'react'
import renderer from 'react-test-renderer'
import {ThemeProvider} from 'styled-components'

import {Header} from 'view/Shared/Header'
import {theme} from 'view/theme'

test('[SNAP] Header', () => {
  const expected = renderer.create(
    <ThemeProvider theme={theme}>
      <Header />
    </ThemeProvider>
  ).toJSON()
  expect(expected).toMatchSnapshot()
})
