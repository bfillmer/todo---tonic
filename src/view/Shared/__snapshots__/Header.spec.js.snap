// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAP] Header 1`] = `
<header
  className="sc-htpNat bGitmu"
>
  <h1
    className="sc-bdVaJa ldpKxe"
  >
    T&T
  </h1>
</header>
`;
